dissect/target/plugins/os/windows/everything/everything_types.py

```diff
@@ -0,0 +1,551 @@
+            # and can be fixed by reading and discarding a single byte here.
+
+            if new_byte_count := read_byte_or_4(self.fh):
+                trunc_from_prev = read_byte_or_4(self.fh)
+                if trunc_from_prev > len(temp_buf):
+                    raise ValueError(f"Error while parsing folder names {trunc_from_prev} > {len(temp_buf)}")
+                temp_buf = temp_buf[: len(temp_buf) - trunc_from_prev]
+            temp_buf += self.fh.read(new_byte_count)
+
+            # The yield can't happen here, because we can only call resolve_path once we finish this loop
+            self.parse_folder(folder, temp_buf.decode())
+
+        for folder in folder_list:
+            yield EverythingF(
+                file_path=folder.resolve_path(folder_list),
+                size=folder.size,
+                attributes=folder.attributes,
+                date_created=folder.date_created,
+                date_modified=folder.date_modified,
+                date_accessed=folder.date_accessed,
+                file_type="directory",
+            )
+
+    def parse_files(self, folder_list: list[EverythingIndexObj]) -> Iterator[EverythingF]:
+        temp_buf = b""
+        for _ in range(self.header.number_of_files):
+            # See comment above for explanation of this loop
+            parent_index = c_everything_header.uint32_t(self.fh)
+            if parent_index > self.header.number_of_filesystems + self.header.number_of_folders:
+                raise ValueError(
+                    "Error while parsing file names. Parent index out of bounds "
+                    f"{parent_index} > {self.header.number_of_filesystems + self.header.number_of_folders}"
+                )
+
+            # There MAY be some edge case where this is okay, and the check should be
+            # parent_index < self.number_of_folders + self.total_filesystem_num, but I haven't seen this yet
+            # Is there any way a file has a root filesystem without a folder?
+            if parent_index >= self.header.number_of_folders:
+                raise ValueError("Something weird, this points to filesystem_index")
+
+            # There is a bug loading EFU files, (which causes Everything v1.4.0.704b to crash)
+            # where it tries to read an extra byte, which is not present in the file.
+            # If at any point parsing EFU files fails, it is likely because of this bug,
+            # and can be fixed by reading and discarding a single byte here.
+
+            file_name = folder_list[parent_index].resolve_path(folder_list)
+            if new_byte_count := read_byte_or_4(self.fh):
+                trunc_from_prev = read_byte_or_4(self.fh)
+                if trunc_from_prev > len(temp_buf):
+                    raise ValueError(f"Error while parsing file name {trunc_from_prev} > {len(temp_buf)}")
+                temp_buf = temp_buf[: len(temp_buf) - trunc_from_prev]
+            temp_buf += self.fh.read(new_byte_count)
```

Comment:
I think this is the same reading loop as before, maybe move this to a function in the global scope called `read_truncated_name` or something?